### ISSUE TITLE
Simplify the generic signatures of our JavaFX MVC classes

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/phoenicis#1750]
When implementing JavaFX components for Phoenicis the developer needs to implement the new component according to the MVC pattern:
- **M**odel: The model (called control in JavaFX) contains the input and output data of the component. This information is normally wrapped inside `Property` objects, to allow for subscriptions to object changes.
- **V**iew: The view (called skin in JavaFX) contains the visualization of the component. The visualization consists of a set of (other) JavaFX components which are grouped in a layout.
- **C**ontroller: The controller (has not yet official support in JavaFX, but in our case named *behavior*) represents the feedback logic from the view to the model. Normally the controller adds listeners to the view, which change the model when some view event occurs.

In Phoenicis the different MVC elements are represented by abstract classes, which need to be extended by the component parts:
- `ControlBase<C extends ControlBase<C, S>, S extends SkinBase<C, S>>`: The base class for a model/control class
- `SkinBase<C extends ControlBase<C, S>, S extends SkinBase<C, S>>`: The base class for a view/skin class
- `BehaviorBase<C extends ControlBase<C, S>, S extends BehaviorSkinBase<C, S, B>, B extends BehaviorBase<C, S, B>>`: The base class for a controller/behavior class. When providing a custom behavior, the view/skin class needs to implement the `BehaviorSkinBase<C extends ControlBase<C, S>, S extends BehaviorSkinBase<C, S, B>, B extends BehaviorBase<C, S, B>>` class instead of the `SkinBase` class.

When looking at the to be extended class, it is quite clear that the developer needs to specify quite a few generics. This has both benefits:
- improves the type safety of the implemented component parts

and disadvantages:
- requires in my opinion extensive knowledge about the use of the to be specified generics to ensure that the generics are specified correctly
- leads to the usage of `?` generics, when base classes are used in other skin classes 
- makes it harder to change the skin of a control or the behavior of a skin through extension, because the generics specify the which class the skin or behavior must have.

Therefore I suggest to simplify the generic signatures of our JavaFX MVC base classes:
- `ControlBase`
- `SkinBase<C extends ControlBase>`
- `BehaviorBase<C extends ControlBase, S extends SkinBase<C>>`

This is also more in line with the root base classes `Control` and `SkinBase` provided by JavaFX.

**Comment by plata** [comment]
Is this still valid?

**Comment by madoar** [comment]
Yes. I'm also thinking about removing `BehaviorBase` completely, but I'm not sure yet. I first want to finish the refactoring of the old JavaFX classes to MVC components. Afterwards I want to continue with this PR.

**Comment by plata** [comment]
Ok.
